lib/src/position.dart

```diff
@@ -213,6 +213,208 @@ abstract class Position<T extends Position<T>> {
     return _legalMovesOf(square);
   }
 
+  Move? parseSan(String sanString) {
+    final aIndex = 'a'.codeUnits[0];
+    final hIndex = 'h'.codeUnits[0];
+    final oneIndex = '1'.codeUnits[0];
+    final eightIndex = '8'.codeUnits[0];
+    String san = sanString;
+
+    final firstAnnotationIndex = san.indexOf(RegExp('[!?#+]'));
+    if (firstAnnotationIndex != -1) {
+      san = san.substring(0, firstAnnotationIndex);
+    }
+
+    if (san == "O-O") {
+      Move? move;
+      if (turn == Side.white) {
+        move = NormalMove(from: parseSquare("e1")!, to: parseSquare("g1")!);
+      }
+      if (turn == Side.black) {
+        move = NormalMove(from: parseSquare("e8")!, to: parseSquare("g8")!);
+      }
+      if (!isLegal(move!)) {
+        return null;
+      }
+      return move;
+    }
+    if (san == "O-O-O") {
+      Move? move;
+      if (turn == Side.white) {
+        move = NormalMove(from: parseSquare("e1")!, to: parseSquare("c1")!);
+      }
+      if (turn == Side.black) {
+        move = NormalMove(from: parseSquare("e8")!, to: parseSquare("c8")!);
+      }
+      if (!isLegal(move!)) {
+        return null;
+      }
+      return move;
+    }
+
+    final isPromotion = san.contains("=");
```

Comment:
I'd rather have single quotes string for consistency in the project. I just added a lint rule on main to help with this.